crates/spk-storage/src/walker.rs

```diff
@@ -0,0 +1,1155 @@
+        &mut self,
+        search_package: &'a Option<String>,
+    ) -> Result<&mut Self> {
+        match search_package {
+            Some(package) => {
+                // Parse the string into package identifying parts
+                let (_, (parts, components)) = all_consuming(
+                    ident_parts_with_components::<nom_supreme::error::ErrorTree<_>>,
+                )(package)
+                .map_err(|err| match err {
+                    nom::Err::Error(e) | nom::Err::Failure(e) => {
+                        miette!(
+                            "Parsing ident from '{package}' for repo walker failed: {}",
+                            e.to_string()
+                        )
+                    }
+                    nom::Err::Incomplete(_) => unreachable!(),
+                })?;
+
+                // Set up a filter function for matching the repo, if
+                // any, and package name.
+                let pkg_name = parts.pkg_name.to_string();
+                self.with_package_filter(move |wp| {
+                    RepoWalkerFilter::exact_package_name_filter(wp, None, pkg_name.clone())
+                });
+
+                // Set up a filter function matching for the version, if any
+                if let Some(version_number) = parts.version_str {
+                    self.with_version_filter(move |version| {
+                        RepoWalkerFilter::exact_match_version_filter(
+                            version,
+                            version_number.to_string().clone(),
+                        )
+                    });
+                }
+
+                // Set up a filter function for matching the build ident, if any
+                if let Some(build_id) = parts.build_str {
+                    self.with_build_ident_filter(move |b| {
+                        RepoWalkerFilter::exact_match_build_digest_filter(
+                            b,
+                            build_id.to_string().clone(),
+                        )
+                    });
+                }
+
+                // Set up a filter function for matching the components, if any
+                if !components.is_empty() {
+                    self.with_component_filter(move |c| {
+                        RepoWalkerFilter::allowed_components_filter(c, &components)
+                    });
+                }
+            }
+            None => {
+                // Leave the filtering functions as they are.
+            }
+        }
+        Ok(self)
+    }
+
+    /// Given some file path string, this will use it to set a file
+    /// filter function that matches the file path against the start
+    /// of each file found by the walker.
+    ///
+    /// If given None instead of a path string, this will not change
+    /// the file filter function.
+    ///
+    /// This is a helper function used by the spk du command. The same
+    /// kind of filter could be setup directly by calling
+    /// with_file_filter().
```

Comment:
Any code references in docs should be in this form so that they actually generate out as links in the rust docs. This will also make sure that whatever you are references actually exists in case refactoring changes the name, etc

```suggestion
    /// [`Self::with_file_filter`].
```

Reply:
I've done a pass to update these.